### PR TITLE
Adds support for external reminder notifications

### LINKED ISSUE #760: Allow external applications to send SMS
External services should be able to `/notify` Gambit with a NS User ID & a message to send.

- [ ] Add routes file for notifications
- [ ] Find the phone number for given NS user
- [ ] Send the given message to the found NS user

### PR DESCRIPTION
**Pull request by deadlybutter**
#### What's this PR do?
Adds an endpoint for triggering reminder notifications 

#### How should this be reviewed?
- hit endpoint w/ missing params 
- hit endpoint w/ all params but missing reminder on campaign
- hit endpoint w/ all params & set reminder msg on campaign

screenshot_20161130-144245

#### Relevant tickets
Fixes #760

_Still need to update docs for this, not ready for merge_

### CODE REVIEW

**Comment by deadlybutter** [comment]
@sergii-tkachenko ready for review

File: app/routes/notifications.js [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+  const reminderType = req.body.reminderType;
+
+  if (!northstarUserId || !campaignId || !reminderType) {
+    res.json({ error: 'Missing parameters' });
+    return;
+  }
+
+  app.locals.clients.northstar.Users
+  .get('id', northstarUserId)
+  .then(nsUser => nsUser.mobile)
```
Comment (sergii-tkachenko) [inline app/routes/notifications.js]: Why do you need this then? Why not just do everything in the next one?
Like

```js
 .then((nsUser) => { // eslint-disable-line consistent-return
  if (!nsUser.mobile) return res.json({ error: 'No mobile number for NS user.' });
  const mobile = nsUser.mobile;
  // ...
```

**Comment by deadlybutter** [comment]
Retested with new parameter missing, set & missing in DB + set in DB.

File: app/routes/notifications.js [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+
+    const reminderMessage = app.locals.campaigns[campaignId][`msg_relative_reminder_${reminderType}`];
+    if (!reminderMessage) return res.json({ error: 'No reminder set for campaign.' });
+
+    mobilecommons.send_message(mobile, reminderMessage);
+    stathat('Sent relative reminder');
+    res.json({ success: true });
+  })
+  .catch((err) => {
+    logger.error('Error sending reminder message', err);
```
Comment (sergii-tkachenko) [inline app/routes/notifications.js]: err.message

File: documentation/endpoints/notifications.md [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+```
+
+Returns either an `error` or `success` object.
+
+**Parameters**
+
+Name | Type | Description
+--- | --- | ---
+`northstarUserId` | `string` | Northstar User Id to send notification for.
+`campaignId` | `string::int` | The campaign ID this notification is for.
```
Comment (sergii-tkachenko) [inline documentation/endpoints/notifications.md]: `string::int`? Is there a spec for this format?

File: app/routes/notifications.js (hunk 1 above)
Comment (deadlybutter) [inline app/routes/notifications.js]: true

File: documentation/endpoints/notifications.md (hunk 3 above)
Comment (deadlybutter) [inline documentation/endpoints/notifications.md]: I have no idea what format this is, traditionally I would do `|` but that breaks the markdown table.

**Comment by deadlybutter** [comment]
@sergii-tkachenko lint errors fixed, good to go

**Comment by aaronschachter** [comment]
> Still need to update docs for this, not ready for merge

@deadlybutter Are there docs? If so, can you link to them?

**Comment by sergii-tkachenko** [comment]
@aaronschachter https://github.com/DoSomething/gambit/blob/develop/documentation/endpoints/notifications.md

File: lib/mobilecommons.js [hunk 4]
```diff
@@ -59,3 +62,28 @@ exports.profile_update = function (profileId, phone, oip, updateFields) {
...
+    form: {
+      phone_number: phoneNumber,
+      body: message,
+      campaign_id: masterCampaignid,
+    },
+  };
+
+  const requestRetry = new RequestRetry();
+  requestRetry.setRetryConditions([400, 408, 500]);
+  requestRetry.post(`${uri}/send_message`, data, (error) => {
```
Comment (aaronschachter) [inline lib/mobilecommons.js]: Hi @sergii-tkachenko @jamjensen @classicfreddie @deadlybutter @justkika  -- just wanted to make sure it's intentional that we're sending Broadcast Declined and Relative Reminder messages as SMS, not MMS. Is the desired behavior?  If they have copy set to over 255 characters, they will be truncated...

Comment (deadlybutter) [inline lib/mobilecommons.js]: If MMS makes more sense than lets do that!

**Pull request merged**